components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1905,14 +1908,47 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
             return;
         }
         self.register_merge_check_tick();
+        fail_point!(
+            "on_check_merge_not_1001",
+            self.fsm.peer_id() != 1001,
+            |_| {}
+        );
         if let Err(e) = self.schedule_merge() {
-            info!(
-                "failed to schedule merge, rollback";
-                "region_id" => self.fsm.region_id(),
-                "peer_id" => self.fsm.peer_id(),
-                "err" => %e,
-            );
-            self.rollback_merge();
+            if self.fsm.peer.is_leader() {
+                self.fsm
+                    .peer
+                    .add_want_rollback_merge_peer(self.fsm.peer_id());
+                if self.fsm.peer.want_rollback_merge_peers.len()
+                    >= raft::majority(
```

Comment:
How to ensure peers in `want_rollback_merge_peers` matches the current `peer_list`?

Reply:
It can be ensured. If a peer can be added to `want_rollback_merge_peers`, it must be in current `peer_list` because its `PrepareMerge` commit index is the same as the leader's.